setup.py

```diff
@@ -46,6 +46,8 @@ def find_version(*file_paths):
     'cylc-flow>=8.0b0',
     'jupyterhub==1.3.*',
     'tornado==6.1.*',
+    'traitlets>=5.0.5',
+    'graphene>=2.1.8',
```

Comment:
I believe setuptools also installs dependencies in order. So this graphene gets installed, then I **think** if the version doesn't match with the version used in `graphene-tornado` below, setuptools may decide to use the version from `graphene-tornado`.

Reply:
Good point. Damn, this dependencies stuff is more involved than I thought :grimacing:

Reply:
As these are transitive dependencies it may be safer to specify them with no version constraint and bump them to the end of the list (which is effectively the same as not specifying them in the `setup.py` file).

Reply:
Ah, good to know. I thought specifying after the dependency would trigger setuptools to install the latest. So +1 to moving them to the end of the file with no version range 👍

Reply:
@kinow and @oliver-sanders - am I misunderstanding what a "transitive dependency" is, or have you missed my comment above?  https://github.com/cylc/cylc-uiserver/pull/198#issuecomment-809834016  

(I'm saying they are explicit dependencies, because we import them directly, not transitive!)

Reply:
Perhaps more a misuse of "transitive" on my part.

They are deps which are brought in by other deps, we don't care much what version we get so long as those deps are happy. At the moment not specifying them as deps at all works fine but is also yuck since we do make some light use of them ourselves.

Reply:
>  we do make some light use of them ourselves 

Well technically that means they are not transitive, so we should specify them as explicit cylc-uiserver deps, right?  Otherwise our code can break implicitly if JupyterHub (say) changes its deps under the hood.

Reply:
Yes.

Reply:
>  So this graphene gets installed, then I *think* if the version doesn't match with the version used in graphene-tornado below, setuptools may decide to use the version from graphene-tornado.

Actually, that would be an unsolvable conflict wouldn't it?  (Two packages request mutually incompatible versions of graphene).

Reply:
Oh, you probably mean not conflicting, i.e. if the version ranges overlap which one will get priority (within the ranges).  But technically that shouldn't matter, because either package should be happy with any version within their specified range.

Reply:
OK I've change the order, to put the most specific requirement first:
```
    'graphene-tornado==2.6.*',
    'graphene>=2.1.8',
```

Reply:
(reverting - I forgot actually look at what version of graphene graphene-tornado wanted!)